sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -224,6 +224,10 @@ def _transform(self, X, fitting):
                                                fitting=fitting,
                                                transforming=transforming,
                                                indices=indices, values=values)
+                else:
+                    raise TypeError(f'Unsupported value Type {type(v)} '
+                                    f'for {f}: {v}.\n'
+                                    f'{type(v)} objects are not supported.')
```

Comment:
Maybe we can actually merge this with the TypeError raised when `v` is an instance of Mapping above. Wonder if one of the core devs has a reason why Mapping was specifically given an elif statement, was it a common for user's to think that `DictVectorizer` supported Mapping instances?

Also would be good to have an extra test case for this since this might have been a [regression](https://github.com/scikit-learn/scikit-learn/issues/19489#issuecomment-783187967)

Reply:
The commit can be found here: https://github.com/scikit-learn/scikit-learn/commit/320b4c61f97fec3facc3c4c2b4cf9351d3425b44

The check for `Mapping` is to prevent nested dictionaries. We can merge this error if we update the check for `Iterable`:

```py
elif not isinstance(v, Mapping) and isinstance(v, Iterable):
    ...
else:
    raise TypeError(...)
```